### PR TITLE
IConfiguration Extension Methods Refactor and Tests

### PR DESCRIPTION
**Pull request by igorquintaes**
This pull request refactors and simplifies configuration value retrieval logic throughout the codebase, focusing on the `GetValues` extension method and its usage. The changes remove unnecessary parsing logic, improve error handling, and add comprehensive unit tests to ensure correctness.

### Refactoring and simplification of configuration methods

* Refactored `GetValues<T>` in `ConfigurationExtensions.cs` to remove the need for a custom parser function; it now directly deserializes configuration sections to lists of type `T`, ensuring distinctness and handling nulls gracefully. Also improved error handling in `GetRequiredValueAsResult<T>` and simplified value conversion logic.
* Updated all usages of `GetValues<T>` across service factories and step services to use the new, simplified method signature, removing redundant enum parsing. [[1]](diffhunk://#diff-ab04bddaa9b5d18b241f2c525fecbc82e024f1d538a59125469dfde1b458deb0L35-R35) [[2]](diffhunk://#diff-075117436e58ecdffe5eff24c9dacd30d71b61ba34226cd9c95b59159716ac85L18-R18) [[3]](diffhunk://#diff-099a456e7301f02c7efe1cb09cc99554ef6801e5b379d927581a38ed1cada7a6L23-R23) [[4]](diffhunk://#diff-d85ca2f41cbcd8a31ce37248680a48f4d59b92b20a24873f20c617b7452a8899L17-R17) [[5]](diffhunk://#diff-dbef60244d1e0c529bc0620028d8ae3381d05a920552fee5bcce23b8e3f73be0L14-R14) [[6]](diffhunk://#diff-3784875f9de8a30b4843c959e7115c3ca06ce8eb7bab4c84ce9c6de174cd3977L13-R13) [[7]](diffhunk://#diff-c711d7ae115c2b46143c6b409c00d44efbdc67a3513a5fb035079d45d66484c2L19-R20)

### Testing improvements

* Added a new test suite `ConfigurationExtensionsTests` that thoroughly tests the updated configuration extension methods, including edge cases for missing keys, invalid values, and duplicate entries.
* Improved test helper methods in `FixtureExtensions.cs` to return the frozen `IConfiguration` instance, supporting the new test scenarios and making tests more expressive. [[1]](diffhunk://#diff-56bc4226b5a5bad9a429917823ddef6cc15e6dd38bf31e463fe55c25e2edddd5L66-L82) [[2]](diffhunk://#diff-56bc4226b5a5bad9a429917823ddef6cc15e6dd38bf31e463fe55c25e2edddd5R98-R99)

### Minor codebase cleanup

* Minor formatting and namespace import adjustments for consistency. [[1]](diffhunk://#diff-c711d7ae115c2b46143c6b409c00d44efbdc67a3513a5fb035079d45d66484c2R7) [[2]](diffhunk://#diff-56bc4226b5a5bad9a429917823ddef6cc15e6dd38bf31e463fe55c25e2edddd5R6)

These changes collectively make configuration retrieval more robust, maintainable, and easier to test.

### CODE REVIEW

File: BotDeScans.App/Extensions/ConfigurationExtensions.cs [hunk 1]
```diff
@@ -12,37 +12,28 @@ public static T GetRequiredValue<T>(this IConfiguration configuration, string ke
...
+            var convertedValue = (T)converter.ConvertFromInvariantString(value)!;
+            return Result.Ok(convertedValue);
         }
-        catch (NotSupportedException)
+        catch (Exception)
         {
-            return Result.Fail($"'{key}' config value contains a not supported value.");
+            var errorMessage = $"'{key}' config value contains an unsupported value.";
+            return Result.Fail(errorMessage);
         }
```
[the remark this answers is not in the document]
Comment (igorquintaes) [reply BotDeScans.App/Extensions/ConfigurationExtensions.cs:30]: As it can use any parser, even customs, we can't predict which exception we should use. Also, we don't want a ex. be thrown, just return a fail result if be unable to parse it - and the final user just must fill the config as doc ask.

File: BotDeScans.UnitTests/Specs/Extensions/ConfigurationExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        public void GivenNotFoundSectionShouldReturnEmpty()
+        {
+            const string SECTION = "SomeSection";
+
+            A.CallTo<IConfigurationSection?>(() => fixture
+                   .FreezeFake<IConfiguration>()
+                   .GetSection(SECTION))
+                   .Returns(null);
+
+            var result = fixture
```
[the remark this answers is not in the document]
Comment (igorquintaes) [reply BotDeScans.UnitTests/Specs/Extensions/ConfigurationExtensionsTests.cs]: b287e5b

File: BotDeScans.UnitTests/Specs/Extensions/ConfigurationExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            var expectedResult = new StepName[]
+            {
+                StepName.Download,
+                StepName.Compress
+            };
+
+            fixture.FreezeFakeConfiguration(SECTION, values)
+                   .GetValues<StepName>(SECTION)
+                   .Should().BeEquivalentTo(expectedResult);
+        }
```
[the remark this answers is not in the document]
Comment (igorquintaes) [reply BotDeScans.UnitTests/Specs/Extensions/ConfigurationExtensionsTests.cs:149]: b287e5b
